oracle/parser.go

```diff
@@ -71,6 +72,45 @@ func (u *Updater) Parse(ctx context.Context, r io.ReadCloser) ([]*claircore.Vuln
 		if len(vs) == 0 {
 			return nil, fmt.Errorf("could not determine dist")
 		}
+
+		// Check if the vulnerability only affects a userspace_ksplice package.
+		//  These errata should never be applied to a container since ksplice
+		//  userspace packages are not supported to be run within a container.
+		// If we couldn't find a CPE list, make sure to include the
+		//  vulnerability. We'd rather have false positives for
+		//  userspace_ksplice packages than have false negatives for
+		//  *everything*.
+		isOnlyKsplice := len(def.Advisory.AffectedCPEList) > 0
+		// If there's at least one ksplice CPE and not all the affected CPEs
+		//  are ksplice related, this will cause false positives we can catch.
+		//  This should (almost) never happen.
+		atLeastOneKsplice := false
+		for _, affected := range def.Advisory.AffectedCPEList {
+			wfn, err := cpe.Unbind(affected)
+			if err != nil {
+				// Found a CPE but could not parse it. Assume it's not a
+				//  userspace_ksplice package.
+				zlog.Warn(ctx).Str("cpe", affected).Msg("could not parse CPE")
+				isOnlyKsplice = false
+				atLeastOneKsplice = true
+				break
+			}
+			if wfn.Attr[cpe.Edition].V == "userspace_ksplice" {
+				atLeastOneKsplice = true
+			} else {
+				isOnlyKsplice = false
+			}
+		}
+
+		if !isOnlyKsplice {
+			//  This should (almost) never happen.
+			if atLeastOneKsplice {
+				zlog.Warn(ctx).Str("def.Title", def.Title).Msg("potential false positives: vulnerability has at least one unskippable ksplice match")
+			}
+		} else {
+			return nil, fmt.Errorf("skippable ksplice vulnerabilities")
```

Comment:
```suggestion
			return nil, errors.New("skippable ksplice vulnerabilities")
```

actually, how come we want to error here? Would it not be valid to `return nil, nil`, or do we prefer the signal an error message?

Reply:
I thought it might be dangerous to return a nil pointer to the caller if there's no error, but I just tested that with a list of pointers in Go Playground, and it doesn't seem to be a problem. I'll add a debug log instead.